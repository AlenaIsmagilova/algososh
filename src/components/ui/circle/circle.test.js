import { Circle } from "./circle";
import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";

describe("test circle ui-component", () => {
  it("circle without text", () => {
    const circleWithoutText = renderer.create(<Circle />);
    expect(circleWithoutText).toMatchSnapshot();
  });

  it("circle with text", () => {
    const circleWithText = renderer.create(<Circle letter="letter" />);
    expect(circleWithText).toMatchSnapshot();
  });

  it("circle with head", () => {
    const circleWithHead = renderer.create(<Circle head="head" />);
    expect(circleWithHead).toMatchSnapshot();
  });

  it("circle with react-element in head", () => {
    const circleWithReactElHead = renderer.create(<Circle head={<Circle />} />);
    expect(circleWithReactElHead).toMatchSnapshot();
  });

  it("circle with tail", () => {
    const circleWithTail = renderer.create(<Circle tail="tail" />);
    expect(circleWithTail).toMatchSnapshot();
  });

  it("circle with react-element in tail", () => {
    const circleWithReactElTail = renderer.create(<Circle tail={<Circle />} />);
    expect(circleWithReactElTail).toMatchSnapshot();
  });

  it("circle with index", () => {
    const circleWithIndex = renderer.create(<Circle index={0} />);
    expect(circleWithIndex).toMatchSnapshot();
  });

  it("circle with props isSmall", () => {
    const circleWithIsSmall = renderer.create(<Circle isSmall={true} />);
    expect(circleWithIsSmall).toMatchSnapshot();
  });

  it("circle with state default", () => {
    const circleDefault = renderer.create(
      <Circle state={ElementStates.Default} />
    );
    expect(circleDefault).toMatchSnapshot();
  });

  it("circle with state modified", () => {
    const circleModified = renderer.create(
      <Circle state={ElementStates.Modified} />
    );
    expect(circleModified).toMatchSnapshot();
  });

  it("circle with state changing", () => {
    const circleChanging = renderer.create(
      <Circle state={ElementStates.Changing} />
    );
    expect(circleChanging).toMatchSnapshot();
  });
});
