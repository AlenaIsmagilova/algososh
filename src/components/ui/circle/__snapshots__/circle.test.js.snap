// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test circle ui-component circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="test-circle_head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with props isSmall 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle  small default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with react-element in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="test-circle_head"
      />
      <div
        className="circle   default"
        data-testid="test-circle_wrapper"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="test-circle_value"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="test-circle_indexValue"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="test-circle_tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with react-element in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="test-circle_head"
      />
      <div
        className="circle   default"
        data-testid="test-circle_wrapper"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="test-circle_value"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="test-circle_indexValue"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="test-circle_tail"
      />
    </div>
  </div>
</div>
`;

exports[`test circle ui-component circle with state changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   changing"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with state default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with state modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   modified"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="test-circle_tail"
  >
    tail
  </div>
</div>
`;

exports[`test circle ui-component circle with text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;

exports[`test circle ui-component circle without text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="test-circle_head"
  />
  <div
    className="circle   default"
    data-testid="test-circle_wrapper"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="test-circle_value"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="test-circle_indexValue"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="test-circle_tail"
  />
</div>
`;
